prod/westeurope/internal/api/functions_assets_r3/function_assets/terragrunt.hcl

```diff
@@ -0,0 +1,115 @@
+  config_path = "../../cosmosdb/database"
+}
+
+dependency "storage_account" {
+  config_path = "../../storage/account"
+}
+
+dependency "storage_account_assets" {
+  config_path = "../../../../common/cdn/storage_account_assets"
+}
+
+dependency "storage_account_logs" {
+  config_path = "../../../../operations/storage_account_logs/account"
+}
+
+# Internal
+dependency "resource_group" {
+  config_path = "../../../resource_group"
+}
+
+# Common
+dependency "virtual_network" {
+  config_path = "../../../../common/virtual_network"
+}
+
+dependency "application_insights" {
+  config_path = "../../../../common/application_insights"
+}
+
+dependency "key_vault" {
+  config_path = "../../../../common/key_vault"
+}
+
+# Include all settings from the root terragrunt.hcl file
+include {
+  path = find_in_parent_folders()
+}
+
+terraform {
+  source = "git::git@github.com:pagopa/io-infrastructure-modules-new.git//azurerm_function_app?ref=v2.1.0"
+}
+
+inputs = {
+  name                = "assets"
+  resource_group_name = dependency.resource_group.outputs.resource_name
+
+  resources_prefix = {
+    function_app     = "fn3"
+    app_service_plan = "fn3"
+    storage_account  = "fn3"
+  }
+
+  app_service_plan_info = {
+    kind     = "elastic"
+    sku_tier = "ElasticPremium"
+    sku_size = "EP1"
+  }
+
+  runtime_version = "~3"
+
+  application_insights_instrumentation_key = dependency.application_insights.outputs.instrumentation_key
+
+  app_settings = {
+    FUNCTIONS_WORKER_RUNTIME       = "node"
+    WEBSITE_NODE_DEFAULT_VERSION   = "10.14.1"
+    WEBSITE_RUN_FROM_PACKAGE       = "1"
+    FUNCTIONS_WORKER_PROCESS_COUNT = 4
+    NODE_ENV                       = "production"
+
+    COSMOSDB_URI  = dependency.cosmosdb_account.outputs.endpoint
+    COSMOSDB_KEY  = dependency.cosmosdb_account.outputs.primary_master_key
+    COSMOSDB_NAME = dependency.cosmosdb_database.outputs.name
+    // TODO: Rename to STORAGE_CONNECTION_STRING
+    QueueStorageConnection = dependency.storage_account.outputs.primary_connection_string
+
+    AssetsStorageConnection    = dependency.storage_account_assets.outputs.primary_connection_string
+    
+    // Keepalive fields are all optionals
+    FETCH_KEEPALIVE_ENABLED             = "true"
+    FETCH_KEEPALIVE_SOCKET_ACTIVE_TTL   = "110000"
+    FETCH_KEEPALIVE_MAX_SOCKETS         = "40"
+    FETCH_KEEPALIVE_MAX_FREE_SOCKETS    = "10"
+    FETCH_KEEPALIVE_FREE_SOCKET_TIMEOUT = "30000"
+    FETCH_KEEPALIVE_TIMEOUT             = "60000"
+
+    SLOT_TASK_HUBNAME = "ProductionTaskHub"
+
+    // Disable functions
+    #"AzureWebJobs.GetService.Disabled"                             = "1"
+    #"AzureWebJobs.GetVisibleServices.Disabled"                     = "1"
+    #"AzureWebJobs.GetServiceByRevision.Disabled"                   = "1"
+  }
+
+  app_settings_secrets = {
+    key_vault_id = dependency.key_vault.outputs.id
+    map = {
+      
+    }
+  }
+
+  allowed_subnets = [
+    dependency.subnet.outputs.id
+  ]
```

Comment:
I think this should be removed as the function must be reachable by the assets CDN